iis-workflows/docs/workflow_conventions.markdown

```diff
@@ -0,0 +1,111 @@
+
+    (output_person_age,  X/Y)
+
+This notation is necessary when using Java MapReduce workflow nodes with multiple outputs.
+
+# Workflow name
+
+The name of the workflow should correspond to the path leading to this workflow in the code repository after removing constant prefix `eu/dnetlib/iis` and removing constant suffix `oozie_app/workflow.xml`. For example, if the path to the workflow is `eu/dnetlib/iis/transformers/citationmatching/oozie_app/workflow.xml`, the entry in the Oozie XML should be
+
+    <workflow-app xmlns="uri:oozie:workflow:0.2" name="transformers_citationmmatching">.
+
+where the slashes in the path are replaced with underscores (`_`). This approach assures that the names of workflows are both unique and quite short.
+
+In case of a test workflow, its name should be created in the same way as the normal workflow but it should be prefixed additionally with `test-`, e.g. the test workflow placed at `eu/dnetlib/iis/transformers/citationmatching/sampledataproducer/oozie_app/workflow.xml` should be called `test-transformers_citationmatching_sampledataproducer`.
+
+# Special types of Oozie actions
+
+We came to use some conventions related to names of actions:
+
+- Prefixed with `skip-` - a node that is used to generate empty data store as an alternative to some "real" processing.
+- Prefixed with `transformers_` or `transformer_` - a node that converts the data stores to a format expected by their consumers.
+- Suffixed with `_collapser` - a node that uses "collapser" infrastructure to de-duplicate records from data store.
+- Suffixed with `_union` - a node that executes a "union" operation on data stores.
+
+# Directory structure corresponding to Oozie action
+
+The **names of Oozie actions** should be valid HDFS file/directory names. This is because a directory named as the action is created in HDFS for most actions.
+
+By default, all data produced by a given workflow node is stored in a directory having the same name as the name of the workflow node. For example, for a node "sample_processor" that produces two output data stores with names "processed_person", "processed_document" we have the following **directory structure**:
+
+    sample_processor/
+    |
+    |- working_dir/
+    |- processed_person/
+	\- processed_document/
+
+
+Here:
+
+- `workingDir` corresponds to a directory where the node can store results of its intermediate computations, temporary files etc. In case of this node being a subworkflow, this is the place where directories of its child workflow nodes are stored. This directory is named `working_dir` in the file system.
+- The "sample_processor" directory is recreated in the `<prepare>` section of the Oozie's action. This is to ensure that the workflow can be properly restarted in case of transient errors and possibly stopped and resumed (I'm not sure about the latter two -- this has to be checked in practice).
```

Comment:
I understand that instead of deleting the whole "sample_processor" directory, we delete the included "working_dir" and output data store directories individually. Why is that? Isn't deleting "sample_processor" directory enough to solve the problem?